SingularityService/src/main/java/com/hubspot/singularity/resources/DeployResource.java

```diff
@@ -134,4 +151,30 @@ public SingularityRequestParent cancelDeploy(
 
     return fillEntireRequest(requestWithState);
   }
+
+  @POST
+  @Path("/update")
+  @ApiOperation(value="Update the target active instance count for a pending deploy", response=SingularityRequestParent.class)
+  @ApiResponses({
+    @ApiResponse(code=400, message="Deploy is not in the pending state pending or is not not present")
+  })
+  public SingularityRequestParent updatePendingDeploy(@ApiParam(required=true) SingularityUpdatePendingDeployRequest updateRequest) {
+    SingularityRequestWithState requestWithState = fetchRequestWithState(updateRequest.getRequestId());
+
+    authorizationHelper.checkForAuthorization(requestWithState.getRequest(), user, SingularityAuthorizationScope.WRITE);
+
+    Optional<SingularityRequestDeployState> deployState = deployManager.getRequestDeployState(requestWithState.getRequest().getId());
+
+    if (!deployState.isPresent() || !deployState.get().getPendingDeploy().isPresent() || !deployState.get().getPendingDeploy().get().getDeployId().equals(updateRequest.getDeployId())) {
+      throw badRequest("Request %s does not have a pending deploy %s", updateRequest.getRequestId(), updateRequest.getDeployId());
+    }
+
+    if (updateRequest.getTargetActiveInstances() > requestWithState.getRequest().getInstancesSafe()) {
+      throw badRequest("Cannot update pending deploy to have more instances (%s) than instances set for request (%s)");
+    }
```

Comment:
we should also check to make sure nobody is setting `targetActiveInstances` to something less than zero